src/sources/kubernetes_logs/mod.rs

```diff
@@ -126,6 +137,20 @@ pub struct Config {
     /// How long to delay removing entries from our map when we receive a deletion
     /// event from the watched stream.
     delay_deletion_ms: usize,
+
+    /// Time period in which the group limit has to be satisfied before resetting
+    /// the counters
+    rate_window_secs: usize,
+
+    /// Rate limiting rules specified by the user to form groups
+    rules: Option<Vec<GroupRule>>,
+}
```

Comment:
We'll need to update the cue docs for the new configuration options as well

Reply:
I also wonder if we want the `rate_window_secs` name to be more tied to the `rules` option since it only affects that operation?

Reply:
A rate window is constraining on the implementation, too high or too low will give poor performance and the user has no way to guess what the "best" window is, if there even is such a thing. I suggest configuring a simple _average rate_ in lines/sec or bytes/sec and then let the implementation do the best it can to maintain that each time it has read or write events to process. The read/write intervals are the optimum windows, and the user can't predict those.